### PR TITLE
Add SQLAlchemy and repoze.sendmail to integration documents

### PR DESCRIPTION
**Pull request by jamadden**
@jimfulton Like these?

### CODE REVIEW

**Comment by jimfulton** [comment]
Yes, thanks!

File: docs/integrations.rst [hunk 1]
```diff
@@ -6,5 +6,18 @@ package so that their transactions can be integerated with others.
 
 `ZODB <http://www.zodb.org>`_
   ZODB was the original user of the ``transaction`` package.  Its
-  transactions are controlled by by ``transaction`` and ZODB fullt
+  transactions are controlled by by ``transaction`` and ZODB fully
```
Comment (jimfulton) [inline docs/integrations.rst]: doh, thanks :)

**Comment by jimfulton** [comment]
It would be good to mention the limitations of these.

From a quick reading:

- The sqlalchemy integration doesn't leverage sqlalchemy's two-phase commit integration.

  For example, my guess is that a Postgres conflict error would be raised in the second phase of TPC.

- Similarly, repoze.sendmail doesn't do it's work till the second phase.

I would refer to these are partial implementations, but our language for this isn't established.

Maybe it would be best to merge this and take a crack at the changes myself.

**Comment by jamadden** [comment]
I'm checking a list of other projects we use to see if I can find anymore. If I do I'll add them. When I'm done, I'll merge.

FWIW, I believe there is an option in zope.sqlalchemy to to configure two-phase commit. I think it's off by default.

**Comment by jimfulton** [comment]
See: http://docs.sqlalchemy.org/en/latest/orm/session_transaction.html

I haven't read deeper than that.

It isn't enough to "configure TPC". The phase-specific interactions would need to be wired up to the transaction package so that the data manager advanced in lock step with the transaction.

**Comment by jimfulton** [comment]
Thanks for researching other projects!

**Comment by jimfulton** [comment]
Oh, sorry, I misread your comment.

I didn't see this in zope.sqlalchemy. Why would one *ever* want it to be turned off?

**Comment by jimfulton** [comment]
Ah, I see. I guess you'd tell sqlalchemy to create a two-phase session and then do the dance. Hm.

(This is from a light reading of the code, which is all I have time for ATM.)

**Comment by jamadden** [comment]
I think that's right, that's what I found our code doing, and [I see that zope.sqlalchemy](https://github.com/zopefoundation/zope.sqlalchemy/blob/master/src/zope/sqlalchemy/datamanager.py#L208) does different things if something looks like two-phase.

**Comment by jamadden** [comment]
At least in our code, twophase is off by default. I don't remember why, but it seems like we had some difficulty with it on.

**Comment by jamadden** [comment]
I didn't find anything else that could be considered outside the direct ZODB ecosystem. There are lots of zope.* things that use transaction in one way or another, but they almost all also use persistent.

**Comment by jimfulton** [comment]
Thanks for looking!

**Pull request merged**